MTV-1727: Improve behavior for Deleting VMs from plan

jschuler opened the pull request:
## 📝 Links

https://issues.redhat.com/browse/MTV-1727

## 📝 Description

Followup to the ticket after JIRA comments.

1. Update wording to consistently say Delete and not Remove (UX docs approved)
2. Fix an old bug that after selecting and Cancelling/Deleting a vm, the button was still enabled even though nothing is checked. Now the button will be disabled again until a user selects a row item.
3. Tooltip now says that deleting the last row item is not allowed.

## 🎥 Demo

https://github.com/user-attachments/assets/e6d3679a-9e43-414b-9fef-689270ef6e09


## 📝 CC://

@sgratch

avivtur commented on packages/forklift-console-plugin/src/components/page/StandardPageWithSelection.tsx:
```diff
@@ -56,7 +56,8 @@ export function withHeaderSelection<T>({
   const Enhanced = ({ dataOnScreen, ...other }: TableViewHeaderProps<T>) => {
     const selectableItems = dataOnScreen.filter(canSelect);
-    const allSelected = selectableItems.every((it) => isSelected(it));
+    const allSelected = selectableItems.length > 0 && selectableItems.every((it) => isSelected(it));
```
`... selectableItems.every((it) => isSelected(it));` can be just `selectableItems.every(isSelected)`

avivtur commented on packages/forklift-console-plugin/src/modules/Plans/views/details/tabs/VirtualMachines/components/PlanVMsDeleteButton.tsx:
```diff
@@ -43,7 +46,7 @@ export const PlanVMsDeleteButton: FC<{
     <ToolbarItem>
       <VMsActionButton onClick={onClick} disabledReason={reason}>
-        {t('Remove virtual machines')}
+        {t('Delete VMs from plan')}
```
why changed from virtual machines to VMs and added from plan?

jschuler replied:
Remove fear of deletion and make it clear it's just affecting the plan.
Shortcut to VMs so that the button is not too long.
Btw we run text changes by UXD copy editors, but I'm open to feedback/changes

sgratch commented:
@jschuler 
Worth keeping both actions with similar labels/colors for buttons and popup text:
![Screenshot from 2025-03-06 18-24-22](https://github.com/user-attachments/assets/5f0c7252-9a1d-4e54-af5c-fd99498ca9ab)
![Screenshot from 2025-03-06 18-24-59](https://github.com/user-attachments/assets/c7ce7670-cf6c-4f35-9a12-9d14ce141c44)
![Screenshot from 2025-03-06 18-28-41](https://github.com/user-attachments/assets/f8ae137b-e41a-4ea8-b224-cb46730b0052)
![Screenshot from 2025-03-06 18-28-26](https://github.com/user-attachments/assets/16e413c3-c6e3-4389-b5d9-34d8418df60e)

jschuler commented:
@sgratch do you mean making the "Delete VMs from plan" button danger style, and the "Cancel virtual machines" button primary style in the VMs tab? I think that would look off and bring too much attention to the button

<img width="842" alt="Screenshot 2025-03-06 at 1 38 06 PM" src="https://github.com/user-attachments/assets/a600ceef-cdf8-4fda-b53f-8639d8fc2de9" />

sgratch commented:
@jschuler 
no, I mean that let's try to keep both action modals look similar, if possible:

1) Both action buttons can stay as there are today (secondary variant), but the labels should be aligned:
"Cancel virtual machines"
"Delete virtual machines"

2) Within  `MigrationVMsCancelModal` and `PlanVMsDeleteModal` - 
titles should both end/not end with a question mark, 
confirm button variant should be the same, 
confirm button label should be similar, e.g. "Delete", "Cancel"

You can add more text to explain what it means to remove/delete vms within the modal body, if you want.

sgratch commented:
CC:// @RichardHoch @anarnold97

jschuler commented:
@sgratch good callouts, but I think we're moving goal posts here. Let's focus on merging what addresses the original issue, and we can make further improvements as followups / separate tickets?

sgratch commented:
> @sgratch good callouts, but I think we're moving goal posts here. Let's focus on merging what addresses the original issue, and we can make further improvements as followups / separate tickets?

Sure.
Please just consider the comments for the delete vms button label in line 49 and open a new bug for the other issues.

jschuler commented:
@sgratch I believe I've addressed all your comments! I changed the primary button in the Cancel modal just to "Cancel" based on your comment, but then I realized both the primary and secondary buttons would just say "Cancel". So I went back to what it was before.

| Before    | After |
| -------- | ------- |
|  <img width="597" alt="delete_before" src="https://github.com/user-attachments/assets/54cf5981-0db8-41e5-ba16-f9aa90c3372d" />  |  <img width="580" alt="delete_after" src="https://github.com/user-attachments/assets/8ba613bc-0678-4613-b59c-b7b2f473727b" />  |
|  <img width="587" alt="cancel_before" src="https://github.com/user-attachments/assets/b67fd40a-b5c6-415a-a271-a2a49257602a" /> | <img width="583" alt="cancel_after" src="https://github.com/user-attachments/assets/1367cc6e-4697-4fb2-be21-241c1a4e0408" />  |

RichardHoch commented:
@jschuler @sgratch The English is fine. But the wording of the second line of "Cancel virtual machines migration" doesn't add any information or a warning the way confirmation messages usually do.

Maybe something like "The other virtual machines will be migrated as planned." or "To migrate the deleted virtual machines later, create a new migration plan."  Could be an implied warning, a simple explanation, or a possible "Next step" . You know the customer request better than I do.
